debezium-connect-rest-extension/src/main/java/io/debezium/kcrestextension/DebeziumResource.java

```diff
@@ -92,20 +93,20 @@ else if (m.lookingAt()) {
         throw new IllegalArgumentException("Invalid version string: \"" + version + "\"");
     }
 
-    private static <T> void addConnectorPlugins(List<ConnectorDescriptor> connectorPlugins, Collection<PluginDesc<T>> plugins) {
+    private static <T> void addConnectorPlugins(Map<String, ConnectorDescriptor> connectorPlugins, Collection<PluginDesc<T>> plugins) {
         plugins.stream()
                 .filter(p -> SUPPORTED_CONNECTORS.contains(p.pluginClass().getName()))
-                .forEach(p -> connectorPlugins.add(new ConnectorDescriptor(p.pluginClass().getName(), p.version())));
+                .forEach(p -> connectorPlugins.put(p.pluginClass().getName() + "#" + p.version(), new ConnectorDescriptor(p.pluginClass().getName(), p.version())));
     }
 
     private synchronized void initConnectorPlugins() {
         if (null == this.availableConnectorPlugins || this.availableConnectorPlugins.isEmpty()) {
             // TODO: improve once plugins are allowed to be added/removed during runtime by Kafka Connect, @see org.apache.kafka.connect.runtime.rest.resources.ConnectorPluginsResource
-            final List<ConnectorDescriptor> connectorPlugins = new ArrayList<>();
+            final Map<String, ConnectorDescriptor> connectorPlugins = new HashMap<>();
             Herder herder = getHerder();
             addConnectorPlugins(connectorPlugins, herder.plugins().sinkConnectors());
             addConnectorPlugins(connectorPlugins, herder.plugins().sourceConnectors());
-            this.availableConnectorPlugins = Collections.unmodifiableList(connectorPlugins);
+            this.availableConnectorPlugins = Collections.unmodifiableList(new ArrayList<>(connectorPlugins.values()));
```

Comment:
Would it make more sense to leverage a `Set<ConnectorDescriptor>` with a `Comparable` interface to control de-duplication rather than using a Map?

Reply:
Wouldn't that require iterating the whole Set in case of adding an item that is not yet in the list? Then this temporary map is the fastest way, no?

Reply:
It depends on the `Set` implementation used. `HashSet`, for example,  uses `HashMap` internally to store it’s elements and  offers performance of order O(1) for insertion, removal and retrieval operations. The same performance applies to `LinkedHashSet` but permits to maintain the inserted order. If you instead need to maintain a certain order defined by a `Comparator`, you need to use the `TreeSet` that has gives performance of order O(log(n)) for insertion, removal and retrieval operations. 

In that case I think you can go with `HashSet`.

Reply:
Thanks @mfvitale.

Given that you're effectively using the `Map` in this case like `HashSet` does under the hood, that's precisely why I wonder why we simply didn't opt for `Set` in this case as it would have been a much cleaner way to implement the behavior without any meaningful loss in performance @rk3rn3r.

Reply:
Ah, for sure. As you know from the last team call, I had only last week and just did it quickly to fix it in the allowed time frame of 1 week. I also created the new, working, slim UI container image and didn't give this not so much thought as usual here, but for sure, this is correct. I think `LinkedHashSet` is even a good way to maintain ordering and with some time now I should have chosen this one.